### DISCUSSION TITLE
[PageBundle] overrite Transformer

### DISCUSSION THREAD

**Discussion by eerison** [sonata-project/SonataAdminBundle#7979] [Q&A] [answered]
As Transformer is final in 4.x the only way to override this class is decorating or is there other way?
https://github.com/sonata-project/SonataPageBundle/blob/4.x/src/Entity/Transformer.php

**Comment by VincentLanglet** [comment, marked as answer]
I would say that most of the time the answer is you can implements you own class, or decorating the existing one.
That's exactly the purpose of decoration.

**Comment by eerison** [comment]
OK just to make sure that don't have some thing like AdminExtension
